src/cli/commands/_execute-lifecycle-script.js

```diff
@@ -15,11 +15,21 @@ export async function execFromManifest(config: Config, commandName: string, pkg:
   }
 
   const cmd: ?string = pkg.scripts[commandName];
-  if (cmd) {
+  if (cmd && isValidCommand(commandName, cmd)) {
     await execCommand(commandName, config, cmd, cwd);
   }
 }
 
+function isValidCommand(commandName: string, cmd: string): boolean {
+  // prevent infinite loop (see: https://github.com/yarnpkg/yarn/issues/1227)
+  // blocks `yarn`, `yarn i`, `yarn install`, with any flags as well
+  if (commandName.endsWith('install') && /^yarn(\si(nstall)?)?(\s\-.*)?$/.test(cmd)) {
```

Comment:
Yes, I understand that. I was leaving it more open in case there are future exceptions that need to be made. I can always change it now to just return the opposite of the conditional and could revisit this pattern in the future when another condition is needed.